PostgreSQL unique constraint violation when editing an entry

### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

Linux Mint Debian 12

### Checked with the latest development build (copy version output from About dialog)

- [x] I made a backup of my libraries before testing the latest development version.
- [x] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Open bib file and the entry editor
2. Change some field (it happened to me when editing the "Publisher" field, but could be other fields as well)
3. At some point you should observe an exception (see below) in the command line
4. Doesn't seem to have relevant effects ...


### Appendix

...
<details>

<summary>Log File</summary>

```
2024-11-09 10:22:37 [pool-2-thread-1] org.jabref.logic.search.indexing.BibFieldsIndexer.insertField()
ERROR: Could not add an entry to the index.: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO bib_fields."yks1w6neg84c" ("entryid", "field_name", "field_value_literal", "field_value_transformed")
VALUES (('00229933'), ('publisher'), ('Springer'), ('Springer'))
 was aborted: ERROR: duplicate key value violates unique constraint "yks1w6neg84c_pkey"
  Detail: Key (entryid, field_name)=(00229933, publisher) already exists.  Call getNextException to see other errors in the batch.
	at org.jabref.merged.module@6.0.6556/org.postgresql.jdbc.BatchResultHandler.handleCompletion(Unknown Source)
	at org.jabref.merged.module@6.0.6556/org.postgresql.core.v3.QueryExecutorImpl.execute(Unknown Source)
	at org.jabref.merged.module@6.0.6556/org.postgresql.jdbc.PgStatement.internalExecuteBatch(Unknown Source)
	at org.jabref.merged.module@6.0.6556/org.postgresql.jdbc.PgStatement.executeBatch(Unknown Source)
	at org.jabref.merged.module@6.0.6556/org.postgresql.jdbc.PgPreparedStatement.executeBatch(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.logic.search.indexing.BibFieldsIndexer.insertField(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.logic.search.indexing.BibFieldsIndexer.updateEntry(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.logic.search.IndexManager$8.call(Unknown Source)
	at org.jabref@6.0.6556/org.jabref.gui.util.UiTaskExecutor$1.call(Unknown Source)
	at javafx.graphics@23.0.1/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "yks1w6neg84c_pkey"
  Detail: Key (entryid, field_name)=(00229933, publisher) already exists.
	at org.jabref.merged.module@6.0.6556/org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(Unknown Source)
	at org.jabref.merged.module@6.0.6556/org.postgresql.core.v3.QueryExecutorImpl.processResults(Unknown Source)
	... 15 more
```

</details>

--- a/src/main/java/org/jabref/logic/search/indexing/BibFieldsIndexer.java
+++ b/src/main/java/org/jabref/logic/search/indexing/BibFieldsIndexer.java
@@ -285,8 +285,10 @@ public class BibFieldsIndexer {
     }
 
     public void updateEntry(BibEntry entry, Field field) {
-        removeField(entry, field);
-        insertField(entry, field);
+        synchronized (entry.getId()) {
+            removeField(entry, field);
+            insertField(entry, field);
+        }
     }
 
     private void insertField(BibEntry entry, Field field) {

